### DISCUSSION TITLE
Field is not defined in row error

### DISCUSSION THREAD

**Discussion by flowbru** [spatie/laravel-ciphersweet#51] [General]
Hi there,
I have a User Model with follwing fields:
`['first_name',
        'last_name',
        'email',
        'password',
        'phone_number',
        'current_organization_id',
        'is_active']`

And there first_name, last_name, email, phone_number are encrypted like so:
`public static function configureCipherSweet(EncryptedRow $encryptedRow): void
    {
        $encryptedRow
            ->addField('email')
            ->addBlindIndex('email', new BlindIndex('email_index'));

        $encryptedRow
            ->addField('first_name')
            ->addBlindIndex('first_name', new BlindIndex('first_name_index'));

        $encryptedRow
            ->addField('last_name')
            ->addBlindIndex('last_name', new BlindIndex('last_name_index'));

        $encryptedRow
            ->addField('phone_number')
            ->addBlindIndex('phone_number', new BlindIndex('phone_number_index'));

    }`

All works good so far but when I try to query the data and do not select all encrypted fields like so:
`User::query()
->isOrganizationMember($organization)
->isClient()
->select('id', 'first_name', 'last_name', 'email')
    ->get();`

Then I get the following error:
`ParagonIE \ CipherSweet\ Exception\ EmptyFieldException
PHP 8.0.30
9.52.12
Field is not defined in row: phone_number`

Is there a way to still only select the needed columns in the query?

Thanks in advance!

**Comment by antonio-bp** [comment]
I have same issue.

I fixed it (temporarly) with something like:

```php
// $query = SomeModel::query()->...

$model = $query->getModel();
if ($model instanceof CipherSweetEncrypted
    && method_exists($model, 'bootUsesCipherSweet')
) {
    $fieldsEncrypted = $model::$cipherSweetEncryptedRow->listEncryptedFields();
    $query->select($fieldsEncrypted);
}

// ...
```

**Comment by johnkmedeiros** [comment]
in the config file, (mine is at api/config/ciphersweet.php) change the key `pemit_empty` to true:

```
    /**
     * The provided code snippet checks whether the $permitEmpty property is set to false
     * for a given field. If it is not set to false, it throws an EmptyFieldException indicating
     * that the field is not defined in the row. This ensures that the code enforces the requirement for
     * the field to have a value and alerts the user if it is empty or undefined.
     * Supported: "true", "false"
     */
    'permit_empty' => false
```

source: https://github.com/spatie/laravel-ciphersweet/pull/43
